src/Conversion/ONNXToTorch/NN/PaddingOp.cpp

```diff
@@ -110,79 +107,144 @@ class ONNXConstantPadNdOpToTorchLowering : public ConversionPattern {
+      for (unsigned int i = 0; i < last_non_zero + 1; i++) {
+        auto f0 = IntegerAttr::get(ty, (dimArray[i].dim_start));
+        Value p0v = rewriter.create<ConstantIntOp>(loc, f0);
+        translatepadsList.push_back(p0v);
+
+        auto f1 = IntegerAttr::get(ty, (dimArray[i].dim_end));
+        Value p1v = rewriter.create<ConstantIntOp>(loc, f1);
+        translatepadsList.push_back(p1v);
+      }
+
+      /*
+      int k = 0;
+      for (unsigned int i = 0; i < intValues.size(); i = i + 2) {
+        auto f0 = IntegerAttr::get(ty, (dimArray[k].dim_start));
+        Value p0v = rewriter.create<ConstantIntOp>(loc, f0);
+        translatepadsList.push_back(p0v);
+        k++;
+      }
+      // read the onnx pad values from array(dim_end values)
+      k = 0;
+      for (unsigned int i = 0; i < intValues.size(); i = i + 2) {
+        auto f1 = IntegerAttr::get(ty, (dimArray[k].dim_end));
+        Value p1v = rewriter.create<ConstantIntOp>(loc, f1);
+        translatepadsList.push_back(p1v);
+        k++;
+        }*/
+    }
 
-    auto f33 = IntegerAttr::get(ty, three);
-    auto f11 = IntegerAttr::get(ty, one);
-    auto f22 = IntegerAttr::get(ty, two);
-    auto f00 = IntegerAttr::get(ty, zero);
+    TensorType data_tensor_type = data.getType().cast<TensorType>();
 
-    Value f3v = rewriter.create<ConstantIntOp>(loc,f33);
-    Value f2v = rewriter.create<ConstantIntOp>(loc,f22);
-    Value f1v = rewriter.create<ConstantIntOp>(loc,f11);
-    Value f0v = rewriter.create<ConstantIntOp>(loc,f00);
+    llvm::outs() << "data type "
+                 << "\n"
+                 << data_tensor_type.getElementType() << "\n"
+                 << "\n";
+    llvm::outs() << "data "
+                 << "\n"
+                 << data << "\n"
+                 << "\n";
 
-    TensorType data_tensor_type  = data.getType().cast<TensorType>();
-    TensorType const_val_tensor_type  = const_value.getType().cast<TensorType>();
+    auto dataTy = Torch::ValueTensorType::get(context,
+        data_tensor_type.getShape(), data_tensor_type.getElementType());
 
-    llvm::outs() << "data type " << "\n" << data_tensor_type.getElementType() << "\n" << "\n";
-    llvm::outs() << "data " << "\n" << data << "\n" << "\n";
+    auto dtt = rewriter.create<torch::TorchConversion::FromBuiltinTensorOp>(
+        loc, dataTy, data);
 
-    auto dataTy = Torch::ValueTensorType::get(context, data_tensor_type.getShape(), 
-								data_tensor_type.getElementType());
-    auto constValTy = Torch::ValueTensorType::get(context, const_val_tensor_type.getShape(), 
-								const_val_tensor_type.getElementType());
+    DenseElementsAttr valueAttr =
+        getDenseElementAttributeFromONNXValue(const_value);
+    auto valueIt = valueAttr.getValues<FloatAttr>().begin();
+    auto valueFloat = (*valueIt).cast<FloatAttr>().getValueAsDouble();
+    auto fval = FloatAttr::get(mlir::FloatType::getF64(context), valueFloat);
+    auto ctt = rewriter.create<ConstantFloatOp>(loc, fval);
```

Comment:
The constant was not being extracted (the value `10` was always being filled in). Is there a utility to unpack a single value when the `dense` is known to contain only a single value (instead of this code I copied from somewhere else)? And what is the approach for failing if the `constant_value` is different than we expect?

Reply:
I can look into this.

Reply:
@ljfitz, @jcamacho1234,

Is this issue resolved?  If so, can you please mark it as resolved?

Reply:
This PR is currently blocked on this comment alone.

@ljfitz and @jcamacho1234, if you can resolve this one, then this PR is good to go for a merge.